js/index.js

```diff
@@ -1,7 +1,76 @@
-// Iteration 1: Names and Input
+/*Iteration 1: Names and Input
+1.1 Create a variable `hacker1` with the driver's name.
+1.2 Print `"The driver's name is XXXX"`.
+1.3 Create a variable `hacker2` with the navigator's name.
+1.4 Print `"The navigator's name is YYYY"`.*/
 
+var hacker1 = "AAA";
+console.log(`The driver's name is ${hacker1}.`);
+var hacker2 = "BBBB";
+console.log(`The navigators's name is ${hacker2}.`);
 
-// Iteration 2: Conditionals
+/*Iteration 2: Conditionals
+2.1. Depending on which name is longer, print:
+- The driver has the longest name, it has XX characters. or
+- It seems that the navigator has the longest name, it has XX characters. or
+- Wow, you both have equally long names, XX characters!.*/
 
+if (hacker1.length > hacker2.length) {
+  console.log("The driver has the longest name, it has " + hacker1.length + "characters.");
+  } else if (hacker1.length < hacker2.length) {
+    console.log("It seems that the navigator has the longest name, it has " + hacker2.length + " characters.");
+  } else {
+    console.log("Wow, you both have equally long names, " + hacker1.length + " characters!");
+};
 
-// Iteration 3: Loops
\ No newline at end of file
+/*Iteration 3: Loops
+3.1 Print all the characters of the driver's name, separated by a space and in capitals i.e. "J O H N"*/
+
+var output3dot1 = "";
+for (var i = 0; i < hacker1.length; i++){
+ output3dot1 += hacker1[i].toUpperCase() + " ";
+}
+console.log(output3dot1);
+
+/*3.2 Print all the characters of the navigator's name, in reverse order. i.e. "nhoJ"*/
+
+var output3dot2 = "";
+for (var i = hacker2.length-1; i >= 0; i--){
+ output3dot2 = output3dot2 + hacker2[i];
+}
+console.log(output3dot2);
+
+/*3.3 Depending on the lexicographic order of the strings, print:
+- The driver's name goes first.
+- Yo, the navigator goes first definitely.
+- What?! You both have the same name?*/
+
+if (hacker1 > hacker2) {
+  console.log("The driver's name goes first.")
+} else if (hacker1 < hacker2) {
+  console.log("Yo, the navigator goes first definitely.")
+} else {
+  console.log("What?! You both have the same name?")
+};
+
```

Comment:
The code works perfectly, but it evaluates if it's in the reverse alphabetical order. To get it in the alphabetical order, you should just need to check if (hacker1 < hacker2) in the first condition, and if (hacker1 > hacker2) in the second.